File: src/main/java/com/sat/common/event/EventsInitializer.java
package com.sat.common.event;

import lombok.RequiredArgsConstructor;
import org.springframework.beans.factory.InitializingBean;
import org.springframework.context.ApplicationContext;
import org.springframework.stereotype.Component;

@RequiredArgsConstructor
@Component
public class EventsInitializer implements InitializingBean{

    private final ApplicationContext applicationContext;

    @Override
    public void afterPropertiesSet() {
        Events.setPublisher(applicationContext);
    }
}

File: src/main/kotlin/com/sat/common/utils/event/EventsInitializer.kt
package com.sat.common.utils.event

import org.springframework.beans.factory.InitializingBean
import org.springframework.context.ApplicationContext
import org.springframework.stereotype.Component

@Component
class EventsInitializer(private val applicationContext: ApplicationContext) : InitializingBean {
    override fun afterPropertiesSet() {
        Events.initialize(applicationContext)
    }
}
